### PR TITLE
Integrate formatter in ESLint

### LINKED ISSUE #516: Integrate formatter in ESLint
- Integrate any formatter in Eslint so as to have consistent code everywhere and get rid of the formatting changes in most PRs by contributors

Find any formatter which can be integrated with Eslint or along with it.
Probably we can we prettier plugin of Eslint.

### PR DESCRIPTION
**Pull request by SivaramanJ**
Closes #516 
1. Modified eslintrc to communicate the following
  -  Extend from the recommended Prettier config.
  - Register the Prettier plug-in we installed.
  - Show Prettier errors as errors.

2. I have applied the formatting changes to all the files so that it follows same set of formatting to all the files.

### CODE REVIEW

**Comment by priyanshk20** [comment]
Looks Good to me.

**Comment by Kshashank99** [comment]
Please link issue no. 516, not the PR no. itself so that when it gets merged, the corresponding issue gets closed simultaneously.

**Comment by SivaramanJ** [comment]
> Please link issue no. 516, not the PR no. itself so that when it gets merged, the corresponding issue gets closed simultaneously. image

Done @Kshashank99

**Review by sumitd94 — APPROVED** [review]

**Review by rohan09-raj — COMMENTED** [review]
IMG_20220210_093415
Can you please check why the tests are failed ??

**Review by rohan09-raj — APPROVED** [review]
Good work 🎉, @SivaramanJ

File: .prettierrc [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
```
Comment (ankurnarkhede) [inline .prettierrc:1]: WHy is prettierrc reuired?
We already are importing the default rules by the plugins.

Comment (ankurnarkhede) [reply .prettierrc:1]: And we already have editorconfig.

**Review by ankurnarkhede — CHANGES_REQUESTED** [review]

File: package.json [hunk 2]
```diff
@@ -12,7 +12,8 @@
     "test-unit": "sh scripts/tests/testUnit.sh",
     "test": "yarn lint && yarn test-unit && yarn test-integration",
     "tdd:watch": "sh scripts/tests/tdd.sh",
-    "generate-api-schema": "node utils/generateAPISchema.js"
+    "generate-api-schema": "node utils/generateAPISchema.js",
+    "pretty": "prettier --write \"./**/*.{js,jsx,json}\""
```
Comment (ankurnarkhede) [inline package.json:16]: This is already done by `lint-fix`, right?

**Review by ankurnarkhede — CHANGES_REQUESTED** [review]

File: .prettierrc (hunk 1 above)
Comment (ankurnarkhede) [reply .prettierrc:1]: We just want to use prettier to be consistent with the formatting.
Just add prettierrc to gitignore to get rid of any issues of contributors' prettierrc file getting added to git.. We will be relying on eslintrc. We are not directly using prettier for formatting but via an eslint plugin.

File: package.json (hunk 2 above)
Comment (SivaramanJ) [reply package.json:16]: Okay  I did not notice the lint fix  cmd ? I will remove the prettier cmd

File: .prettierrc (hunk 1 above)
Comment (SivaramanJ) [reply .prettierrc:1]: My intention is to have uniform styles acroos the file,  but did not notice the editor config. Will remove this file as well

**Review by ankurnarkhede — APPROVED** [review]
LGTM ✅ 
Good work 👍

**Pull request merged**